source/Sashimi/AzureWebAppServiceMessageHandler.cs

```diff
@@ -63,9 +65,27 @@ public AzureWebAppServiceMessageHandler(ILog logger)
                 endpoint.WebAppSlotName = webAppSlotName;
             }
 
+            endpoint.DefaultWorkerPoolId = GetWorkerPoolId(messageProperties, variables, workerPoolIdResolver);
+
             return endpoint;
         }
 
+        string GetWorkerPoolId(IDictionary<string, string> messageProperties, VariableDictionary variables, Func<string, string> workerPoolIdResolver)
+        {
+            messageProperties.TryGetValue(AzureWebAppServiceMessageNames.WorkerPoolIdOrNameAttribute, out var workerPoolIdOrName);
+            if (string.IsNullOrWhiteSpace(workerPoolIdOrName))
+                workerPoolIdOrName = variables.Get(KnownVariables.WorkerPool.Id);
```

Comment:
So this is the line where it falls back to using the WorkerPoolId from the running step, if the caller hasn't specified one, right? Could be worth adding a comment here to make that clear for future us.